shared-bindings/_bleio/Adapter.c

```diff
@@ -96,18 +96,24 @@ const mp_obj_property_t bleio_adapter_enabled_obj = {
 };
 
 //|     address: Address
-//|     """MAC address of the BLE adapter. (read-only)"""
+//|     """MAC address of the BLE adapter."""
 //|
 STATIC mp_obj_t bleio_adapter_get_address(mp_obj_t self) {
     return MP_OBJ_FROM_PTR(common_hal_bleio_adapter_get_address(self));
 
 }
 MP_DEFINE_CONST_FUN_OBJ_1(bleio_adapter_get_address_obj, bleio_adapter_get_address);
 
+STATIC mp_obj_t bleio_adapter_set_address(mp_obj_t self, mp_obj_t new_address) {
+    common_hal_bleio_adapter_set_address(self, new_address);
```

Comment:
Check for success:
```c
    if (!common_hal_bleio_adapter_set_address(self, new_address) {
        mp_raise_bleio_BluetoothError(translate("Could not set address"));
    }
```

There may be some `_bleio` implementations where this is not possible, or there's some possible failure mode, so we need to check.